File: scripts/verify_forecasts.js
const mongoose = require('mongoose');
const fs = require('fs');
const path = require('path');

// Helper to load .env.local
function loadEnvLocal() {
    const envPath = path.join(__dirname, '../.env.local');
    if (fs.existsSync(envPath)) {
        const content = fs.readFileSync(envPath, 'utf-8');
        content.split('\n').forEach(line => {
            const parts = line.split('=');
            if (parts.length >= 2) {
                const key = parts.shift().trim();
                let value = parts.join('=').trim();
                value = value.replace(/^["'](.*)["']$/, '$1');
                if (key && value) {
                    process.env[key] = value;
                }
            }
        });
    }
}
loadEnvLocal();

const URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/agri_trend_dashboard';

async function verifyForecasts() {
    try {
        await mongoose.connect(URI, { dbName: 'agri_trend_dashboard' });
        console.log('✅ Connected to DB: agri_trend_dashboard');

        const Forecast = mongoose.connection.db.collection('forecasts');

        const count = await Forecast.countDocuments();
        console.log(`Total Forecasts: ${count}`);

        if (count === 0) {
            console.log('❌ No forecasts found.');
            process.exit(1);
        }

        const models = await Forecast.distinct('model');
        console.log('Models found:', models);

        if (!models.includes('LSTM')) {
            console.log('❌ LSTM forecasts missing.');
        } else {
            console.log('✅ LSTM forecasts present.');
        }

        if (!models.includes('Linear Regression')) {
            console.log('❌ Linear Regression forecasts missing.');
        } else {
            console.log('✅ Linear Regression forecasts present.');
        }

        const sample = await Forecast.findOne();
        console.log('Sample Forecast:', sample);

        process.exit(0);
    } catch (err) {
        console.error('❌ Error:', err.message);
        process.exit(1);
    }
}

verifyForecasts();

File: scripts/verify_forecasts.py
import os
import sys
from pymongo import MongoClient
from dotenv import load_dotenv

BASE_DIR = os.path.dirname(os.path.abspath(__file__))
ENV_PATH = os.path.join(BASE_DIR, '../.env.local')
load_dotenv(ENV_PATH)

def check():
    try:
        URI = os.getenv("MONGODB_URI")
        client = MongoClient(URI)
        db = client.get_database('agri_trend_dashboard')
        
        print("\n🔮 Checking Forecasts Collection...")
        count = db['market_forecasts'].count_documents({})
        print(f"   Total Forecasts: {count}")
        
        if count == 0:
            print("   ⚠️ No forecasts found. Model training might have failed or not run.")
            return

        countries = ["India", "Japan", "Philippines"]
        for c in countries:
            sample = db['market_forecasts'].find_one({"country": c})
            if sample:
                print(f"   ✅ Found forecast for {c}: {sample.get('commodity')} - {sample.get('price')} (Model: {sample.get('model')})")
            else:
                print(f"   ❌ No forecast for {c}")

    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    check()
